### PR TITLE
feat: Generate fraud proof after trigger

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
In an optimint full node, after a fraudulent state transition is detected, a Fraud Proof should be generated.

Uses https://github.com/celestiaorg/tendermint/pull/63 for ABCI interface

Closes: #499

### CODE REVIEW

File: state/executor.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"github.com/celestiaorg/optimint/types"
 )
 
+var fraudProofsEnabled = true
```
Comment (tzdybal) [inline state/executor.go:24]: This shouldn't be global. Please move it to `BlockExecutor` struct.

File: state/executor.go [hunk 2]
```diff
@@ -301,80 +303,130 @@ func (e *BlockExecutor) execute(ctx context.Context, state types.State, block *t
...
+			// Block producer: Initial ISRs generated here
+			block.Data.IntermediateStateRoots.RawRootsList = ISRs
+		}
 	}
 
 	return abciResponses, nil
 }
 
-func (e *BlockExecutor) checkFraudProofTrigger(generatedIsr []byte, currentIsrs [][]byte, index int) error {
+func (e *BlockExecutor) checkFraudProofTrigger(generatedIsr []byte, currentIsrs [][]byte, index int) bool {
```
Comment (tzdybal) [inline state/executor.go:397]: Instead of passing `[][]byte` and `index`, you can pass element of `currentIsrs` directly.

**Review by tzdybal — COMMENTED** [review]
Left few comments. I need to look at this in IDE ;)

File: state/executor.go [hunk 3]
```diff
@@ -268,18 +270,18 @@ func (e *BlockExecutor) execute(ctx context.Context, state types.State, block *t
 	currentIsrs := block.Data.IntermediateStateRoots.RawRootsList
 	currentIsrIndex := 0
 
-	if currentIsrs != nil {
-		expectedLength := len(block.Data.Txs) + 2
-		// BeginBlock + DeliverTxs + EndBlock
-		if len(currentIsrs) != expectedLength {
-			return nil, fmt.Errorf("invalid length of ISR list: %d, expected length: %d", len(currentIsrs), expectedLength)
+	if fraudProofsEnabled {
+		if currentIsrs != nil {
```
Comment (Manav-Aggarwal) [inline state/executor.go:274]: add an else here and halt there

**Comment by tzdybal** [comment]
Replaced by https://github.com/celestiaorg/rollmint/pull/567

**Pull request closed without merging**